server/client.go

```diff
@@ -767,16 +767,23 @@ func (c *client) setPermissions(perms *Permissions) {
 
 // Check to see if we have an expiration for the user JWT via base claims.
 // FIXME(dlc) - Clear on connect with new JWT.
-func (c *client) checkExpiration(claims *jwt.ClaimsData) {
+func (c *client) setExpiration(claims *jwt.ClaimsData, validFor time.Duration) {
+	tn := time.Now().Unix()
```

Comment:
Again, you don't need that for the first "if", so move it just before `expiresAt`

Reply:
fixed